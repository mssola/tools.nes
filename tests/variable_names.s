.segment "HEADER"
    .byte 'N', 'E', 'S', $1A
    .byte $02
    .byte $01
    .byte $12                   ; let's fake working RAM
    .byte $00

.segment "CODE"

zp_good  = $00
zp_bad   = $0300
m_good   = $0200
m_bad    = $02
wr_bad   = $01
wr_good  = $6001
wr_out   = $0400

;; Just so we don't get "unused" warnings
lda zp_bad
lda zp_good
lda m_bad
lda m_good
lda wr_bad
lda wr_good
lda wr_out
