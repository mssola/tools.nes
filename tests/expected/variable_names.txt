warning: you are assigning the zero-page variable 'zp_bad' ($0300) to something outside of zero page (variable_names.s: line 11)
warning: you are assigning the variable 'm_bad' ($02) to something in zero page (variable_names.s: line 13)
warning: you are assigning the variable 'wr_bad' ($01) to something in zero page (variable_names.s: line 14)
warning: you are assigning the variable 'wr_out' ($0400) to something outside of Working RAM (variable_names.s: line 16)
